File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(const char* keymap[], size_t keymap_len, const char* targets[], size_t targets_len) {
    int A[26];

    for(int i=0;i<26;i++){A[i]=101;}
    for(int i=0;i<keymap_len;i++){
        int j=0;
        while(keymap[i][j]){
            int index = keymap[i][j] - 65;
            if(A[index] > j+1){A[index]=j+1;}
            j++;
        }
    }

    int* answer = (int*)malloc(targets_len*sizeof(int));
    for(int i=0;i<targets_len;i++){
        answer[i] = 0;
    }

    for(int i=0;i<targets_len;i++){
        int j=0;
        while(targets[i][j]){
            int index = targets[i][j] - 65;
            if(A[index]>100){answer[i] = -1;break;}
            else{answer[i] += A[targets[i][j] - 65];}
            j++;
        }
    }

    return answer;
}

File: 프로그래머스/1/160586.대충 만든 자판-Python/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    a = [200 for i in range(26)]
    for k in keymap:
        for i in range(len(k)):
            idx = ord(k[i]) - 65
            if a[idx] > i+1:
                a[idx] = i+1

    for t in targets:
        count = 0
        for i in range(len(t)):
            idx = ord(t[i]) - 65
            if a[idx] > 100:
                count = -1
                break
            else:
                count += a[idx]
        answer.append(count)


    return answer
